img_proof/ipa_cloud.py

```diff
@@ -357,8 +360,16 @@ def _run_tests(self, tests, ssh_config):
         print('Arguments:\n{}\n'.format(args))
 
         cmds = shlex.split(args)
-        plugin = Report()
-        result = pytest.main(cmds, plugins=[plugin])
+        num_retries = 0
+
+        while num_retries < self.retry_count:
```

Comment:
... when `self.retry_count == None` ?

Reply:
It can never be None. :)

The value comes from chain map with default value of 3. It's chain mapped here https://github.com/SUSE-Enceladus/img-proof/blob/2b19ce6e4849e94c324ff9f4606b810bdd5987d1/img_proof/ipa_cloud.py#L142. And set here https://github.com/SUSE-Enceladus/img-proof/blob/2b19ce6e4849e94c324ff9f4606b810bdd5987d1/img_proof/ipa_cloud.py#L172 where it's cast to an int.

Reply:
Basically we have a list of dictionaries that get combined based on order of importance. cmd_line -> config -> defaults if there is a value other then none in cmd_line or config then chain map assume it's more important thus the default value wouldn't get used and you'd end up with duplication of defaults. This makes handling any number of configurations very simple. For example with EC2 we also check for values in .ec2utils conf where the order would be cmd_line -> .ec2utils -> config -> defaults etc. I realize chain map is not widely used but it's quite nice in this case to save a lot of repetitive code.